doc/subsystems/bluetooth/hci.txt

```diff
@@ -0,0 +1,994 @@
+its device address and is from a scanner that is allowed by the advertising
+filter policy.
+
+Note: In case Extended Advertising configuration is used for setting up
+advertising sets, this shall not generate any vendor events.
+
+	Status:                                         Size: 1 Octet
+	+--------------------+--------------------------------------+
+	| Value              | Parameter Description                |
+	+--------------------+--------------------------------------+
+	| 0x00               | Command succeeded                    |
+	| 0x01-0xFF          | Command failed                       |
+	+--------------------+--------------------------------------+
+
+	Enable:                                         Size: 1 Octet
+	+--------------------+--------------------------------------+
+	| Value              | Parameter Description                |
+	+--------------------+--------------------------------------+
+	| 0x00               | Scan request reporting disabled      |
+	| 0x01               | Scan request reporting enabled       |
+	| All other values   | Reserved for future use              |
+	+--------------------+--------------------------------------+
+
+When the Set_Scan_Request_Reports command has completed, a Command Complete
+event shall be generated.
+
+
+Zephyr Vendor Events
+====================
+
+The Vendor Event is used to encapsulate all vendor specific events. The event
+code of all vendor events shall be 0xFF. The subevent code is the first octet
+of the event parameters. The subevent code shall be set to one of the valid
+subevent codes from a vendor specific event. All other subevent parameters are
+defined in the vendor specific events.
+
+
+Zephyr Fatal Error
+==================
+
+This event reports a fatal non-recoverable error from the local Controller.
+
++-------------------------------+------------+-------------------------------+
+| Event                         | Event Code | Event Parameters              |
++-------------------------------+------------+-------------------------------+
+| Fatal_Error                   | 0xFF       | Subevent_Code,                |
+|                               |            | Program_Counter,              |
+|                               |            | Error_Info                    |
++-------------------------------+------------+-------------------------------+
+
+The Error_Info provides a string representation of the occurred error. It may
+include information like file name and line numbers.
+
+	Subevent_Code:                                  Size: 1 Octet
+	+--------------------+--------------------------------------+
+	| Value              | Parameter Description                |
+	+--------------------+--------------------------------------+
+	| 0x02               | Subevent code for Fatal Error event  |
+	+--------------------+--------------------------------------+
+
+	Program_Counter:                               Size: 4 Octets
```

Comment:
Good question. Are we targeting 64-bit MCUs eventually. So far all of them are only 32-bit since once you have 64-bit Zephyr might be overkill. But I am fine with making this larger.